tests/jdom/newtrace.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { DEVICE_CHANGE, REFRESH, SensorReg, SystemReg } from "../../src/jdom/constants";
```

Comment:
start new PRs for these workstreams, this should live under the /servers folder.

Reply:
Yeah. What I think I'll do is to break withBus (minus the packet trace replay code, since it's a bit weird and may be obsoleted) and other test utilities into their own PR, add a basic set of non-adapter tests (probably with the monolithic button server), push that for review immediately, then build the CSV-like replay on top of that so it can be tested.

Will also move the adapters mockup to their own PR.